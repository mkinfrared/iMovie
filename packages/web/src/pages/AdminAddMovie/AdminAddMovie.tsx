import React from "react";
import { useHistory } from "react-router-dom";

import AddMovieForm from "components/AddMovieForm";

const AdminAddMovie = () => {
  const { goBack } = useHistory();

  return <AddMovieForm onCancel={goBack} />;
};

export default React.memo(AdminAddMovie);
