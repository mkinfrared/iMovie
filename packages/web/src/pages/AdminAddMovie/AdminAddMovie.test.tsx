import { render } from "@testing-library/react";
import React, { ReactElement } from "react";

import AdminAddMovie from "./AdminAddMovie";

jest.mock("react-router-dom", () => ({
  useHistory: jest.fn(() => ({
    goBack: jest.fn()
  }))
}));

jest.mock("components/AddMovieForm", () => (props: any) => (
  <div data-testid="components/AddMovieForm">
    <code>{JSON.stringify(props)}</code>
  </div>
));

describe("<AdminAddMovie />", () => {
  let Component: ReactElement;

  beforeEach(() => {
    Component = <AdminAddMovie />;
  });

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("should match a snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
