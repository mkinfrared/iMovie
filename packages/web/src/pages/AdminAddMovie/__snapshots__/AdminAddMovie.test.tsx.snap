// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminAddMovie /> should match a snapshot 1`] = `
<div>
  <div
    data-testid="components/AddMovieForm"
  >
    <code>
      {}
    </code>
  </div>
</div>
`;
